test/lib/ansible_test/_internal/sanity/import.py

```diff
@@ -92,91 +103,112 @@ def test(self, args, targets, python_version):
-                message=r['message'],
-                path=os.path.relpath(r['path'], relative_temp_root) if r['path'].startswith(relative_temp_root) else r['path'],
-                line=int(r['line']),
-                column=int(r['column']),
-            ) for r in results]
+            env.update(
+                SANITY_TEMP_PATH=ResultType.TMP.path,
+            )
 
-        results = settings.process_errors(results, paths)
+            if data_context().content.collection:
+                env.update(
+                    SANITY_COLLECTION_FULL_NAME=data_context().content.collection.full_name,
+                    SANITY_EXTERNAL_PYTHON=python,
+                )
+
+            virtualenv_python = os.path.join(virtual_environment_bin, 'python')
+            virtualenv_pip = generate_pip_command(virtualenv_python)
+
+            # make sure requirements are installed if needed
+            if requirements_file:
+                install_cryptography(args, virtualenv_python, python_version, virtualenv_pip)
+                run_command(args, generate_pip_install(virtualenv_pip, 'sanity', context='import-plugins'), env=env, capture=capture_pip)
+
+            # make sure coverage is available in the virtual environment if needed
+            if args.coverage:
+                run_command(args, generate_pip_install(virtualenv_pip, '', packages=['setuptools']), env=env, capture=capture_pip)
+                run_command(args, generate_pip_install(virtualenv_pip, '', packages=['coverage']), env=env, capture=capture_pip)
+
+            try:
+                # In some environments pkg_resources is installed as a separate pip package which needs to be removed.
+                # For example, using Python 3.8 on Ubuntu 18.04 a virtualenv is created with only pip and setuptools.
+                # However, a venv is created with an additional pkg-resources package which is independent of setuptools.
+                # Making sure pkg-resources is removed preserves the import test consistency between venv and virtualenv.
+                # Additionally, in the above example, the pyparsing package vendored with pkg-resources is out-of-date and generates deprecation warnings.
+                # Thus it is important to remove pkg-resources to prevent system installed packages from generating deprecation warnings.
+                run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pkg-resources'], env=env, capture=capture_pip)
+            except SubprocessError:
+                pass
+
+            run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'setuptools'], env=env, capture=capture_pip)
+            run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pip'], env=env, capture=capture_pip)
+
+            display.info(import_type + ': ' + data, verbosity=4)
+
+            cmd = ['importer.py', '--type', import_type]
```

Comment:
Pass `import_type` using an environment variable instead.

```suggestion
            cmd = ['importer.py']
```

Reply:
Implemented in cb7558ad9cc040045e830d28268091345af1abc6.